pkg/sql/codegen/codegen_feature_column.go

```diff
@@ -41,34 +40,37 @@ func isXGBoostModule(module string) bool {
 	return strings.HasPrefix(module, "xgboost")
 }
 
-// MarshalToJSONOrDie converts any data to JSON string.
-// Exit the process if there is any error.
-func MarshalToJSONOrDie(in interface{}) string {
+// MarshalToJSONString converts any data to JSON string.
+func MarshalToJSONString(in interface{}) (string, error) {
 	bytes, err := json.Marshal(in)
-	if err != nil {
-		log.Fatal(err)
-	}
-	return string(bytes)
+	return string(bytes), err
 }
 
 // GenerateFeatureColumnCode generates feature column code for both TensorFlow and XGBoost models
 func GenerateFeatureColumnCode(fc ir.FeatureColumn, module string) (string, error) {
 	switch c := fc.(type) {
 	case *ir.NumericColumn:
+		shapeStr, err := MarshalToJSONString(c.FieldDesc.Shape)
```

Comment:
```go
s, e ;= json.Marshal(c.FieldDesc.Shape)
return fmt.Sprintf("%s", s)  // %s automatically converts []byte into string.
```

Reply:
Same reason why I choose to write a new function as [https://github.com/sql-machine-learning/sqlflow/pull/2433#discussion_r437066025].